### PR TITLE
[FEAT] Build release python wheels and upload to AWS S3

### PR DESCRIPTION
**Pull request by raunakab**
# Overview
New workflow to build commit and upload to AWS S3.
- faster than previous
- uses buildjet
- utilizes build caching (from buildjet)

### CODE REVIEW

**Review by jaychia — APPROVED** [review]

File: .github/actions/install/action.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    shell: bash
+    run: |
+      curl -LsSf https://astral.sh/uv/install.sh | sh
+      UV_BIN="$HOME/.local/bin"
+      echo 'export PATH="$UV_BIN:$PATH"' >> $HOME/.bashrc
+      echo "$UV_BIN" >> $GITHUB_PATH
+  - name: Source changes to add new installs to path
+    shell: bash
+    run: |
+      source $HOME/.bashrc
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/actions/install/action.yaml]: Removed.

File: .github/workflows/build-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        fi
+    - name: Build release wheel
+      run: |
+        if [ "$COMMIT_BUILT" -eq 1 ]; then
+          echo "Commit already built"
+          exit 0
+        fi
+        export CARGO_TARGET_DIR=~/target
+        uv v
+        source .venv/bin/activate
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml:61]: The `uv v` command is a short-form for the `uv venv` command, so what is there is actually fine.

But good suggestion nonetheless! Good bot! You get a tasty treat! Here's a floppy disk: 💾.

File: .github/workflows/build-commit.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        uv pip install pip maturin boto3
+        maturin build --release
+    - name: Upload wheel to AWS S3
+      id: upload
+      run: |
+        if [ "$COMMIT_BUILT" -eq 1 ]; then
+          echo "Commit already built"
+          exit 0
+        fi
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml]: Good catch! Good bot. Here's another treat for you: 💾.

File: .github/workflows/build-commit.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        uv v
+        source .venv/bin/activate
+        uv pip install pip maturin boto3
+        maturin build --release
+    - name: Upload wheel to AWS S3
+      id: upload
+      run: |
+        if [ "$COMMIT_BUILT" -eq 1 ]; then
+          echo "Commit already built"
+          wheel=$(aws s3 ls s3://github-actions-artifacts-bucket/builds/0319d66afe3cf99c8281b1e1dc9cb979cfd4fba6/ | awk '{print $4}')
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml]: Oh wow, good catch. You're on a roll! Here's another treat: 💾.

[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml]: Updated.

File: .github/workflows/build-commit.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml]: Updated.

File: .github/workflows/build-commit.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        maturin build --release
+    - name: Upload wheel to AWS S3
+      id: upload
+      run: |
+        if [ "$COMMIT_BUILT" -eq 1 ]; then
+          echo "Commit already built"
+          wheel=$(aws s3 ls s3://github-actions-artifacts-bucket/builds/${{ github.sha }}/ | awk '{print $4}')
+          echo "wheel=$wheel" >> $GITHUB_OUTPUT
+        else
+          count=$(ls ~/target/wheels/*.whl 2> /dev/null | wc -l)
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/build-commit.yaml:75]: In a previous step, I set the `CARGO_TARGET_DIR` env variable to point to another location (specifically in the home directory). So the current configuration is in fact right.

**Pull request merged**